### PR TITLE
771 psalm config 7 1 x

### PR DESCRIPTION
**Pull request by j6s**
| Q             | A
| ------------- | ---
| Branch        | 7.1.x
| Bug fix?      | no
| New feature?  | no 
| BC breaks?    | no     
| Deprecations? | no 
| Tests pass?   | yes    
| Fixed issues  | chameleon-system/chameleon-system#771   
| License       | MIT

Sister PR to https://github.com/chameleon-system/chameleon-shop/pull/120 - contains the same changes as https://github.com/chameleon-system/chameleon-shop/pull/120 but merged `7.1.x` into the branch and is opened against `7.1.x` in order to prevent merge conflicts.

The only changes between this and #120 are the following 2 commits:

* 9ddf54fab4e7ad8f412424652c281067fc54353b (merge commit)
* 980d47233099ddef166d850f8614fd01515b9dae

### CODE REVIEW

**Review by bestform — CHANGES_REQUESTED** [review]

File: src/SearchBundle/objects/db/TShopSearchCache.class.php [hunk 1]
```diff
@@ -46,6 +46,10 @@ class TShopSearchCache extends TShopSearchCacheAutoParent
      */
     public $aFilter = null;
 
+    /**
+     * @param string $id
+     * @param string $sLanguageId
```
[the remark this answers is not in the document]
Comment (j6s) [reply src/SearchBundle/objects/db/TShopSearchCache.class.php:51]: :heavy_check_mark:

File: src/SearchBundle/objects/db/TShopSearchLog.class.php [hunk 2]
```diff
@@ -11,6 +11,10 @@
 
 class TShopSearchLog extends TShopSearchLogAutoParent
 {
+    /**
+     * @param string $id
+     * @param string $sLanguageId
```
[the remark this answers is not in the document]
Comment (j6s) [reply src/SearchBundle/objects/db/TShopSearchLog.class.php:16]: :heavy_check_mark:

File: src/ShopAffiliateBundle/objects/db/TPkgShopAffiliateList.class.php [hunk 3]
```diff
@@ -31,6 +31,10 @@ protected function &_NewElement(&$aData)
         }
     }
 
+    /**
+     * @param string $sQuery
+     * @param string $sLanguageId
```
[the remark this answers is not in the document]
Comment (j6s) [reply src/ShopAffiliateBundle/objects/db/TPkgShopAffiliateList.class.php:36]: :heavy_check_mark:

File: src/ShopArticleDetailPagingBundle/Service/AddParametersToUrlService.php [hunk 4]
```diff
@@ -25,6 +25,11 @@ public function __construct(UrlUtilityServiceInterface $urlUtility)
         $this->urlUtility = $urlUtility;
     }
 
+    /**
+     * @return string
+     *
+     * @param null|string $url
```
[the remark this answers is not in the document]
Comment (j6s) [reply src/ShopArticleDetailPagingBundle/Service/AddParametersToUrlService.php:31]: :heavy_check_mark:

File: src/ShopBundle/Service/ShopSearchSuggest.php [hunk 5]
```diff
@@ -55,12 +57,15 @@ public function getSearchSuggestions($searchTerm)
             $finalQuery .= '('.$sQuery.')';
             $isFirst = false;
         }
+
+        /** @var array{name:string}[] $results */
```
[the remark this answers is not in the document]
Comment (j6s) [reply src/ShopBundle/Service/ShopSearchSuggest.php:61]: First off: It should actually be `array{name: string, articlenumber: string}`

We are selecting `name` and `articlenumber` from the database. The return value of `fetchAll` will be a sequential array of associative arrays that each contain `name` and `articlenumber` keys - e.g.

```php
[
    [ 'name' => 'foo', 'articlenumber' => 'bar' ],
    [ 'name' => 'foo', 'articlenumber' => 'bar' ],
    [ 'name' => 'foo', 'articlenumber' => 'bar' ],
]
```

Maybe the following would be more immediately recognizable?

```
list<array{name: string, articlenumber: string}>
```

File: src/ShopBundle/mappers/manufacturer/TPkgShopManufacturerMapper_Overview.class.php [hunk 6]
```diff
@@ -79,6 +79,7 @@ protected function getSearchInput()
 
     /**
      * @param TdbShopManufacturerList $oManufacturerList
+     * @param bool|false $bCachingEnabled
```
[the remark this answers is not in the document]
Comment (j6s) [reply src/ShopBundle/mappers/manufacturer/TPkgShopManufacturerMapper_Overview.class.php:82]: :heavy_check_mark:

File: src/ShopBundle/mappers/order/TPkgShopMapper_OrderArticleList.class.php [hunk 7]
```diff
@@ -42,6 +43,7 @@ public function Accept(IMapperVisitorRestricted $oVisitor, $bCachingEnabled, IMa
      * get the value map for the whole article list (order items) of the order.
      *
      * @param TdbShopOrder $oOrder
+     * @param bool|false $bCachingEnabled
```
[the remark this answers is not in the document]
Comment (j6s) [reply src/ShopBundle/mappers/order/TPkgShopMapper_OrderArticleList.class.php:46]: :heavy_check_mark:  I swear I wasn't drinking

File: src/ShopBundle/mappers/order/TPkgShopMapper_OrderPayment.class.php [hunk 8]
```diff
@@ -40,6 +40,7 @@ public function Accept(IMapperVisitorRestricted $oVisitor, $bCachingEnabled, IMa
 
     /**
      * @param TdbShopOrder $oOrder
+     * @param bool|false $bCachingEnabled
```
[the remark this answers is not in the document]
Comment (j6s) [reply src/ShopBundle/mappers/order/TPkgShopMapper_OrderPayment.class.php:43]: :heavy_check_mark:

File: src/ShopBundle/objects/BackendModules/MTShopStatistic/MTShopStatistic.class.php [hunk 9]
```diff
@@ -273,11 +315,28 @@ public function GetHtmlHeadIncludes()
...
+
+    /** @var bool */
     public $bShowGrandTotal = true;
 
+    /**
+     * Returns
+     * - (mixed) The value of the column if it exists
+     * - (int) A sum of the values of subgroups, if it does not but subgroups exist and `bShowGrandTotal=true` is set
+     * - (string) An empty string, if the column does not exist but subgroups exist and `bShowGrandTotal=false` is set
+     * - (int) 0 If the column does not exist and has no subGroups - irresponsible of the value of `bShowGrandTotal`
```
[the remark this answers is not in the document]
Comment (j6s) [reply src/ShopBundle/objects/BackendModules/MTShopStatistic/MTShopStatistic.class.php:335]: :heavy_check_mark:  Yes, that's what I meant

File: src/ShopBundle/objects/WebModules/MTShopBasketCore/MTShopBasketCoreEndpoint.class.php [hunk 10]
```diff
@@ -747,6 +789,8 @@ public function RemoveFromBasket($iArticleId = null, $sConsumer = null, $bIsInte
      * @param string $sBasketItemKey
      * @param string $sConsumer       - optional. if no consumer is give, the response will be send to the global consumer.
      * @param bool   $bIsInternalCall
+     *
+     * @return void
```
[the remark this answers is not in the document]
Comment (j6s) [reply src/ShopBundle/objects/WebModules/MTShopBasketCore/MTShopBasketCoreEndpoint.class.php:793]: :heavy_check_mark: I added them (I didn't add `bool|false` though)

File: src/ShopBundle/objects/WebModules/MTShopOrderWizardCore/MTShopOrderWizardCoreEndPoint.class.php [hunk 11]
```diff
@@ -157,6 +159,8 @@ public static function SetCallingURL($sURL)
      * the value from get/post from self::URL_PARAM_STEP_METHOD = xx.
      *
      * @param string $sStepMethod - method to execute. defaults to ExecuteStep
+     *
+     * @return false|null
```
[the remark this answers is not in the document]
Comment (j6s) [reply src/ShopBundle/objects/WebModules/MTShopOrderWizardCore/MTShopOrderWizardCoreEndPoint.class.php:163]: :heavy_check_mark:

File: src/ShopBundle/objects/db/TShopOrderStep/TShopOrderStepList.class.php [hunk 12]
```diff
@@ -139,6 +141,9 @@ public function GetActiveStepPosition()
     /**
      * render the step list.
      *
+     * @param string $sViewType
+     * @param string $sViewName
+     * @param string $sSpotName
```
[the remark this answers is not in the document]
Comment (j6s) [reply src/ShopBundle/objects/db/TShopOrderStep/TShopOrderStepList.class.php:146]: :heavy_check_mark:

File: src/ShopListFilterBundle/objects/db/TPkgShopListfilterItemList.class.php [hunk 13]
```diff
@@ -15,7 +15,7 @@ class TPkgShopListfilterItemList extends TAdbPkgShopListfilterItemList
      * create a new instance.
      *
      * @param string $sQuery
-     * @param int    $iLanguage
+     * @param string|null $iLanguageId
```
[the remark this answers is not in the document]
Comment (j6s) [reply src/ShopListFilterBundle/objects/db/TPkgShopListfilterItemList.class.php:18]: :heavy_check_mark:

**Pull request merged**